File: CORREDOR/Tipos Constructores/JS/Persona.js
const AccionesPersona = require("./AccionesPersona");

//Constructor Copia
class Persona {
    constructor(nombre, edad){
        this.nombre = nombre;
        this.edad = edad;
    }

    static copiar(otraPersona){
        return new this(otraPersona.nombre, otraPersona.edad);
    }

    mostrarInfo(){
        throw new Error("El método 'mostrarInfo()' debe ser implementado por la clase hija.");
    }
}
module.exports = Persona;

File: CORREDOR/Tipos Constructores/Java/Persona.java
package Java;

// Constructor COPIA que recibe otra Persona
abstract class Persona {
    String nombre;
    String documento;
    
    // Constructor normal
    public Persona(String nombre, String documento) {
        this.nombre = nombre;
        this.documento = documento;
    }

    // Constructor copia
    public Persona(Persona otra) {
        this.nombre = otra.nombre;
        this.documento = otra.documento;
    }

    abstract void mostrarInformacion();
}